pkg/manager/router/router_score.go

```diff
@@ -365,16 +366,18 @@ func (router *ScoreBasedRouter) rebalance(maxNum int) {
 			break
 		}
 		conn := ce.Value
-		router.logger.Info("begin redirect connection", zap.Uint64("connID", conn.ConnectionID()),
-			zap.String("from", busiestBackend.addr), zap.String("to", idlestBackend.addr),
-			zap.Int("from_score", busiestBackend.score()), zap.Int("to_score", idlestBackend.score()))
-		busiestBackend.connScore--
-		router.adjustBackendList(busiestEle)
-		idlestBackend.connScore++
-		router.adjustBackendList(idlestEle)
-		conn.phase = phaseRedirectNotify
-		conn.lastRedirect = curTime
-		conn.Redirect(idlestBackend.addr)
+		if conn.IsRedirectable() {
```

Comment:
It will cause a deadloop: the next time, the connection is chosen again, and again, and again.
I don't think it's very necessary to add this check: `mgr.redirectInfo.Load() == nil` is ensured and closing status is handled.

Reply:
> It will cause a deadloop: the next time, the connection is chosen again, and again, and again.

How? You already filtered it by `conn.phase`.

> I don't think it's very necessary to add this check: mgr.redirectInfo.Load() == nil is ensured and closing status is handled.

True. But I want to leverage these checks into `conn`, instead of tracking them in the router.

Reply:
If you check the `phase` or `redirectInfo` only in `conn`, then it will cause deadloop. If you also check it in the router, you are still tracking it in the router. I don't get your purpose.

Reply:
> If you check the phase or redirectInfo only in conn

I don't get why there will be deadloop.

> If you also check it in the router, you are still tracking it in the router. I don't get your purpose.

I want to check `phase/redirectInfo` in the router by methods of `conn`. Basically I want to move the functionality of `connWrapper` into `conn`, reducing the complexity of router.

It matters when I want to separate router/conn by interfaces/channels, or even RPC.